latch_cli/services/cp/upload.py

```diff
@@ -442,21 +461,23 @@ def end_upload(
         progress_bars.update_total_progress(1)
 
 
-def end_upload_listener(
-    queue: "QueueType",
-    parts_by_src: "PartsBySrcType",
-    upload_info_by_src: "UploadInfoBySrcType",
-):
+def throttler(t: Throttle, q: "LatencyQueueType"):
```

Comment:
This design is weird because it has a separate process that calculates based on the last posted latency.

* We want to use some mean (probably arithmetic) of the last n posted latencies within some timeframe.
* We can probably implement a function called `get_delay` or something that takes the latency queue and returns a number or None
* If we have been blocked a lot recently (like more than 10 seconds), we should probably try the request and fail with a Load error if that does not work.
* Then, we do not need a separate spinning process. The only concurrent object we need is a queue that evicts its oldest object when it overflows some set size.

Reply:
will talk irl